### PR TITLE
parallel-workload: Backup&Restore, multiple DBs, less locking

### PR DESCRIPTION
**Pull request by def-**
General improvements too, see commits.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/python/materialize/parallel_workload/action.py [hunk 1]
```diff
@@ -858,7 +897,12 @@ def run(self, exe: Executor) -> None:
             source_id = self.rng.randrange(len(self.db.kafka_sources))
             source = self.db.kafka_sources[source_id]
             query = f"DROP SOURCE {source}"
-            exe.execute(query)
+            try:
```
Comment (nrainer-materialize) [inline misc/python/materialize/parallel_workload/action.py:988]: Optional: extract
```
            try:
                exe.execute(query)
            except QueryError as e:
                # expected, see #20465
                if self.db.scenario != Scenario.Kill or error_substring not in e.msg:
                    raise e
  ```

**Review by nrainer-materialize — APPROVED** [review]

**Comment by def-** [comment]
This is running into deadlocks, I have to debug further.

File: misc/python/materialize/parallel_workload/parallel_workload.py [hunk 2]
```diff
@@ -48,56 +61,95 @@ def run(
...
+            f"ALTER SYSTEM SET max_roles = {len(databases) * MAX_ROLES * 2}"
+        )
+        system_exe.execute(
+            f"ALTER SYSTEM SET max_clusters = {len(databases) * MAX_CLUSTERS * 2}"
+        )
+        system_exe.execute(
+            f"ALTER SYSTEM SET max_replicas_per_cluster = {MAX_CLUSTER_REPLICAS * 2}"
+        )
+        # Most queries should not fail because of privileges
+        system_exe.execute(
```
Comment (philip-stoev) [inline misc/python/materialize/parallel_workload/parallel_workload.py:118]: You can fold this into a `for object_type in ["TABLES","SECRETS"...]:`

**Review by philip-stoev — APPROVED** [review]

File: misc/python/materialize/parallel_workload/action.py [hunk 3]
```diff
@@ -376,27 +374,61 @@ def run(self, exe: Executor) -> None:
...
+
+class RenameViewAction(Action):
+    def run(self, exe: Executor) -> None:
+        if self.db.scenario != Scenario.Rename:
+            return
+        with self.db.lock:
+            if not self.db.views:
+                return
+            view = self.rng.choice(self.db.views)
+            old_name = str(view)
```
[the remark this answers is not in the document]
Comment (def-) [reply misc/python/materialize/parallel_workload/action.py:411]: This calls __str__ which already calls identifier internally.

File: misc/python/materialize/parallel_workload/action.py [hunk 4]
```diff
@@ -376,27 +374,61 @@ def run(self, exe: Executor) -> None:
...
                 return
             table = self.rng.choice(self.db.tables)
-            old_name = str(table)
-            table.rename += 1
+        old_name = str(table)
+        table.rename += 1
+        try:
+            exe.execute(f"ALTER TABLE {old_name} RENAME TO {identifier(table.name())}")
+        except:
+            table.rename -= 1
```
[the remark this answers is not in the document]
Comment (def-) [reply misc/python/materialize/parallel_workload/action.py]: We keep track of the current name using this counter. If the rename failed, we have to reset it.

File: misc/python/materialize/parallel_workload/action.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nrainer-materialize) [reply misc/python/materialize/parallel_workload/action.py:412]: ok 👍

**Pull request merged**